### PR TITLE
botonic-react: fix cover component in botonic react #BLT-1223

### PR DESCRIPTION
**Pull request by Iru89**
## Description

Create a separate file for the cover component (using typescript and reducing the size of the webchat file).
Add the toggleCoverComponent function in the WebchatContext to avoid having to pass the function using props

## Context

Removing the scroll in a previous PR was not rendering the cover component correctly

## Approach taken / Explain the design

Use a conditional render to render the cover component or the webchat. It is not necessary to render the messages under the cover component.

### CODE REVIEW

File: packages/botonic-react/src/webchat/cover-component/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from 'react'
+
+import { WEBCHAT } from '../../constants'
+import { WebchatContext } from '../../contexts'
+
+interface Props {
```
Comment (vanbasten17) [inline packages/botonic-react/src/webchat/cover-component/index.tsx:6]: rename to `CoverComponentProps`

File: packages/botonic-react/src/webchat/cover-component/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  const closeCoverComponent = () => {
+    toggleCoverComponent(false)
+  }
+
+  if (!Cover) {
+    return null
+  }
+
+  return Cover && webchatState.isCoverComponentOpen ? (
```
Comment (vanbasten17) [inline packages/botonic-react/src/webchat/cover-component/index.tsx:30]: no need to check for `Cover` again if already checked above.

just return `webchatState.isCoverComponentOpen && <Cover ...`
and before `if (!Cover || !webchatState.isCoverComponentOpen`

**Review by vanbasten17 — APPROVED** [review]
Nice it is a starting point for having this in TS, I think we can still improve it in next iterations

File: packages/botonic-react/src/webchat/cover-component/index.tsx (hunk 1 above)
Comment (Iru89) [reply packages/botonic-react/src/webchat/cover-component/index.tsx:6]: CoverComponentProps already exists elsewhere in the code

File: packages/botonic-react/src/webchat/cover-component/index.tsx (hunk 2 above)
Comment (Iru89) [reply packages/botonic-react/src/webchat/cover-component/index.tsx]: applied

File: packages/botonic-react/src/webchat/cover-component/index.tsx (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/cover-component/index.tsx:6]: so we'll need to refactor this in the future, we should not have it twice. Let's do it in next iterations as we simplify the code

**Pull request merged**